dd-java-agent/agent-profiling/profiling-uploader/src/main/java/com/datadog/profiling/uploader/ProfileUploader.java

```diff
@@ -90,14 +91,16 @@
       new Callback() {
         @Override
         public void onFailure(final Call call, final IOException e) {
-          log.warn("Failed to upload profile", e);
+          HttpUrl url = call.request().url();
+          log.warn("Failed to upload profile to {}", url, e);
         }
 
         @Override
         public void onResponse(final Call call, final Response response) {
           if (response.isSuccessful()) {
             log.debug("Upload done");
           } else {
```

Comment:
```suggestion
          } else if (config.getApiKey() == null && response.getStatusCode() == 404) {
```

Or whatever is the equivalent code to get to the statuscode.

If it is deemed to help with readability we could also add a `config.isProfilingAgentless()` function that basically returns the result of `config.apiKey != null`